doc/wol/Wake-on-LAN-HLD.md

```diff
@@ -88,19 +92,56 @@ Offset   |0     |1     |2     |3     |4     |5     |
          +-----------------------------------------+
 ```
 
+### Magic Pattern in UDP Payload
+**Magic Packet** is a Ethernet frame with structure:
+
+* **IP Header**:
+  * **Destination IP**: user provided target IP address, cloud be IPv4 address or IPv6 address, default is `255.255.255.255`.
+* **UDP Header**:
+  * **Destination Port**: user provided target port, default is 9.
+* **UDP Payload**:
+  * Six bytes of all `0xff`. [6 bytes]
+  * Sixteen repetitions of the target device's MAC address. [96 bytes]
+  * (Optional) A four or six byte password. [4 or 6 bytes]
+
+```
+Byte     |
+Offset   |0     |1     |2     |3     |4     |5     |
+---------+------+------+------+------+------+------+
+           \      \      \      \      \      \     
+         +------+------+------+------+------+------+
+         |.. Destination IP(4 bytes or 16 bytes) ..|
+         +------+------+------+------+------+------+      
+           \      \      \      \      \      \      
+         +------+------+---------------------------+
+         |  ... Destination Port(2 bytes) ...      |   
+         +------+------+---------------------------+
+           \      \      \      \      \      \      
+         +------+------+------+------+------+------+------------------------------
+       N | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF |
+         +------+------+------+------+------+------+           UDP
+     N+6 | Target MAC (repeat 16 times, 96 bytes)  |           PAYLOAD
+         +-----------------------------------------+
+   N+102 |    Password (optional, 4 or 6 bytes)    |
+         +-----------------------------------------+
+```
+
 ## CLI Design
 
 The `wol` command is used to send magic packet to target device.
 
 ### Usage
 
 ```
-wol <interface> <target_mac> [-b] [-p password] [-c count] [-i interval]
+wol <interface> <target_mac> [-b] [-u] [-a ip-address] [-t udp-port] [-p password] [-c count] [-i interval]
 ```
 
 - `interface`: SONiC interface name.
 - `target_mac`: a list of target devices' MAC address, separated by comma.
 - `-b`: Use broadcast MAC address instead of target device's MAC address as **Destination MAC Address in Ethernet Frame Header**.
+- `-u`: Let device send magic pattern in udp payload, when this flag was used, -b won't work and destination mac is set by device's networking stack.
+- `-a ip-address`: This parameter only work when -u flag was used. Let device send packet to the ip address, this ip address cloud IPv4 address or IPv6 address.
```

Comment:
What will happen if user specify a list of MAC addresses as `target_mac`, but only a single IP address?

Reply:
We will use the same IP address for them and send packet per mac like we used to do.